distributed/executor/executor.go

```diff
@@ -40,7 +40,7 @@ func NewExecutor(option *ExecutorOption, instructions *pb.InstructionSet) *Execu
 func (exe *Executor) ExecuteInstructionSet() error {
 
 	// start a listener for stats
-	listener, err := net.Listen("tcp", ":0")
+	listener, err := net.Listen("tcp4", ":0")  //exclude ipv6 issue by temporarily using tcp4 only
```

Comment:
Is this needed for the PR?

Reply:
Oh, sorry I didn't clarify this commit. It is not included in the initial PR, but as I made further test on my target environment, the whole flow seemed to pause at some point, although no more error messages comes out. I think the grpc issue is still there, but I've no idea how to debug them with my limited knowledge of it right now. However, I suspected it was due to the ipv6 protocol the system has automatically chosen by specifying `tcp` when you create the executor listener, so I simply force it to use `tcp4` in order to verify my assumption (had to push the code to github.com so that I can build it in my target environment). 

It turns out that using tcp4 has finally solved the issue. But forcing the listener to use tcp4 is by no means the best solution. I guess a better solution would be providing a launch argument such as `useIPV4only`. So you don't have to include this commit in the PR.

On the other hand, there must be something tricky in the OS setting on my problematic environment that interfere the use of IPv6 protocol, I'll see to it when I've got the time.